{
  "WorkItem": {
    "AffectedComponent": {
      "Name": "",
      "DisplayName": ""
    },
    "ClosedComment": "This is indeed a duplicate of the issue and fixed in 0.9.",
    "ClosedDate": "2013-04-10T16:38:49.49-07:00",
    "CommentCount": 1,
    "Custom": null,
    "Description": "I have this sample code, which I expect to compile, but it doesn't. \r\n\r\n__a.ts:__\r\n\r\n```\r\nmodule mine {\r\n    export class A {\r\n    }\r\n}\r\n\r\n```\r\n__b.ts:__\r\n\r\n```\r\n/// <reference path=\"a.ts\" />\r\nmodule mine {\r\n    class P {\r\n    }\r\n\r\n    export class B {\r\n        private b: P[];\r\n    }\r\n}\r\n\r\n```\r\nI get the following error: \r\n\r\nThe name 'P' does not exist in the current scope\r\n\r\nB.TS compiles fine if one of the following changes are made:\r\n1. add export keyword to class P .. i.e., export class P instead of class P\r\n2. Remove the <reference include>\r\n\r\nI believe this is a bug in typescript compiler, so reporting it here.",
    "LastUpdatedDate": "2013-04-10T16:38:49.49-07:00",
    "PlannedForRelease": "",
    "ReleaseVisibleToPublic": false,
    "Priority": {
      "Name": "Unassigned",
      "Severity": 0,
      "Id": 0
    },
    "ProjectName": "typescript",
    "ReportedDate": "2013-04-09T18:21:51.68-07:00",
    "Status": {
      "Name": "Closed",
      "Id": 4
    },
    "ReasonClosed": {
      "Name": "Duplicate"
    },
    "Summary": "Can't resolve types unless it is exported",
    "Type": {
      "Name": "Unassigned",
      "Id": 5
    },
    "VoteCount": 1,
    "Id": 899
  },
  "FileAttachments": [],
  "Comments": [
    {
      "Message": "See also: https://typescript.codeplex.com/workitem/641",
      "PostedDate": "2013-04-10T07:03:26.933-07:00",
      "Id": 96593
    }
  ]
}